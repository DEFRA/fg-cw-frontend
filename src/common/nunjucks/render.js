import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
