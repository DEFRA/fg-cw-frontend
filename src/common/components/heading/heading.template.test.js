import { describe, expect, test } from "vitest";
import { render } from "../../nunjucks/render.js";

describe("heading", () => {
  test("renders", () => {
    const component = render("heading", {
      text: "Services",
      caption: "A page showing available services",
    });

    expect(component).toMatchSnapshot();
  });
});
