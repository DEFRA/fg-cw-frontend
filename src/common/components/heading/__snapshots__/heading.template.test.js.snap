// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`heading > renders 1`] = `
"<div class="govuk-grid-row app-heading" data-testid="app-heading">
  <div class="govuk-grid-column-two-thirds">
    <h1
      class="govuk-heading-xl govuk-!-margin-bottom-2"
      data-testid="app-heading-title"
    >
      Services
    </h1>

      <span class="govuk-caption-m" data-testid="app-heading-caption"
        >A page showing available services</span
      >
  </div>
</div>
"
`;
